code/modules/power/apc/apc_mapping.dm

```diff
@@ -5,5 +5,28 @@
 /obj/machinery/power/apc/auto_name
 	auto_name = TRUE
 
-MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/power/apc/worn_out, APC_PIXEL_OFFSET)
-MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/power/apc/auto_name, APC_PIXEL_OFFSET)
+#define APC_DIRECTIONAL_HELPERS(path)\
+##path/directional/north {\
+	dir = SOUTH; \
+	MAP_SWITCH(pixel_z, pixel_y) = -APC_PIXEL_OFFSET; \
+} \
+##path/directional/south {\
+	dir = NORTH; \
+	MAP_SWITCH(pixel_z, pixel_y) = APC_PIXEL_OFFSET; \
+} \
+##path/directional/east {\
+	dir = WEST; \
+	pixel_x = -APC_PIXEL_OFFSET; \
+	MAP_SWITCH(pixel_z, pixel_y) = 16; \
+} \
+##path/directional/west {\
+	dir = EAST; \
+	pixel_x = APC_PIXEL_OFFSET; \
+	MAP_SWITCH(pixel_z, pixel_y) = 16; \
+}
```

Comment:
I'm not actually sure if this is how it works (GitHub is too slow to let me look back at the defines file all teh way at the top) but is this something that could be X-Macro'd? as in `#define APC_DIRECTIONAL_HELPERS(path) SOME_SHIT(path, -APC_PIXEL_OFFSET, APC_PIXEL_OFFSET, 16, 16)`?

Reply:
yesssssss in theory see above

Reply:
done!